src/FourBitTwoDisclosureDeviceUnlocker.java

```diff
@@ -83,7 +83,7 @@ public static boolean unlock(final Device dev) {
                     break;
                 }
             }
-            n--;
+            n++;
```

Comment:
We clearly need to go the other direction if we do this, and increase n instead of decrease it.